### PR TITLE
fix: Verify that memory and CPU limits are greater than requests

### PR DESCRIPTION
**Pull request by amisevsk**
### What does this PR do?
* Fail workspace and set error message when memory/CPU limit is less than request
* Set memory request to match limit if default request is used and limit is less (e.g. user sets `32Mi` as a limit and no request, use `32Mi` as the request instead of `64Mi`).

### What issues does this PR fix or reference?
Fixes https://github.com/devfile/devworkspace-operator/issues/462

### Is it tested? How?
Added test cases.

### CODE REVIEW

**Comment by benoitf** [comment]
I think it would be interesting to report the error to the user as well if there is a deployment issue ( today it's mangled)

**Review by JPinkney — APPROVED** [review]

**Comment by sleshchenko** [comment]
> I think it would be interesting to report the error to the user as well if there is a deployment issue ( today it's mangled)

DevWorkspace Operator should report deployment issues https://github.com/devfile/devworkspace-operator/issues/465

File: pkg/library/container/testdata/fixes-memory-limit-less-than-default-request.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: "Gives useful error when memory limit is less than default request"
```
Comment (sleshchenko) [inline pkg/library/container/testdata/fixes-memory-limit-less-than-default-request.yaml:1]: Sorry for being late to the party, and that's fully OK you merged without my approval.

It seems to be copy/paste mistake. Is fixed in https://github.com/devfile/devworkspace-operator/pull/467

Comment (amisevsk) [reply pkg/library/container/testdata/fixes-memory-limit-less-than-default-request.yaml:1]: Good catch, I should've waited :smile: 

https://github.com/devfile/devworkspace-operator/pull/469

**Pull request merged**